### ISSUE TITLE
Self Payments for on site

### ISSUE THREAD

**Issue by ianlayton** [MidwestFurryFandom/rams#191]
Kinda waffling a bit on payment options, would like to be able to pay themselves with card at walkin. Anyway to make it so people can pay via card if they aren’t using one of our terminals so we don’t have people taking up terminal space and filling in their credit card info on it?

**Comment by kitsuta** [comment]
Yes, this is actually the default behavior. The option is hidden right now due to the `kiosk_cc_enabled` setting being turned off.

When it's turned on, one of the payment options will say "Pay with credit card now (faster)". If someone selects this when filling out their info, it will direct them to a Stripe payment page where they can fill in their CC info after they register.

While I highly recommend enabling this option, it's not perfect. The most troublesome thing is that, when registering via their phone, people will sometimes close the registration page as soon as they see the green "checkmark" in the stripe popup. This may result in someone paying for their badge and not getting it. It's rare, but significant enough at MAGFest's size that we always have someone on-call who can check Stripe to make those badges after the fact.

Also, while our pre-reg process lets people build up a 'cart' before paying, this doesn't happen during onsite reg. As soon as someone fills their info in, they are asked to pay. This is done to help prevent people from leaving their personal info up on public laptops by accident, but it has its downsides -- I have found via observation that the prime reason people skip self-payment is because they want to pay for multiple badges at once.

**Comment by ianlayton** [comment]
So I'm guessing there's no way to differinate between someone using the kiosk and someone using their phone, correct? Do you find, for MagFest, people paying at the kiosk to be a problem?

**Comment by kitsuta** [comment]
Nope, people do a good job of filling out their info and paying at the kiosk. The problems with phone payments are also very rare, and I think the only year I even heard about it was a year where our internet was incredibly slow.

**Comment by ianlayton** [comment]
The only pause I have about doing payments at convention without touching the card is that we had six membership purchased by two people on the last day of prereg using fake cards. 

The other worry I have, and I thought I brought it up in here, but apparently not, is how hard is it to turn off a membership level on site. It's possible we may run out of shiny and sponsor level gifts during the convention. It easier to change a level before payment is made then have to refund people.

Given that though, I think it would be quicker if we do allow self payment on site and we'll see how it run this year. If we see issues with fake cards being used on site, we may, unfortunately, have to turn that in the future. :/

**Comment by ianlayton** [comment]
After our talk last night and confirmation that we can easily limit number of sponsor badges sold, I think I'm happy with how it's running on staging currently.
